react/modules/extraEvents.ts

```diff
@@ -20,13 +29,16 @@ export async function sendExtraEvents(e: PixelMessage) {
     case 'vtex:userData': {
       const { data } = e
 
+      const criptoID = await digestMessage(data.email)
+
       if (!data.isAuthenticated) {
         return
       }
```

Comment:
Shouldn't we move it past the `isAuthenticated` check? Is there any reason you put it before?
I also think we should be aware that email can be `null`/`undefined` for a number of reasons, and that would make a hash to be generated when it shouldn't. We should prepare for that.

```suggestion
      if (!data.isAuthenticated) {
        return
      }
      
      const criptoID = data.email ? await digestMessage(data.email) : undefined
```

Reply:
@icazevedo Good afternoon.
I made the necessary changes. Excellent observation, thank you.